js/modules/k6/experimental/fs/module.go

```diff
@@ -0,0 +1,315 @@
+}
+
+// Stat returns a promise that will resolve to a [FileInfo] instance describing
+// the file.
+func (f *File) Stat() *goja.Promise {
+	promise, resolve, _ := promises.New(f.vu)
+
+	go func() {
+		resolve(f.file.stat())
+	}()
+
+	return promise
+}
+
+// Read the file's content, and writes it into the provided Uint8Array.
+//
+// Resolves to either the number of bytes read during the operation
+// or EOF (null) if there was nothing more to read.
+//
+// It is possible for a read to successfully return with 0 bytes.
+// This does not indicate EOF.
+func (f *File) Read(into goja.Value) *goja.Promise {
+	promise, resolve, reject := f.vu.Runtime().NewPromise()
+
+	if common.IsNullish(into) {
+		reject(newFsError(TypeError, "read() failed; reason: into argument cannot be null or undefined"))
+		return promise
+	}
+
+	intoObj := into.ToObject(f.vu.Runtime())
+	if !isUint8Array(f.vu.Runtime(), intoObj) {
+		reject(newFsError(TypeError, "read() failed; reason: into argument must be a Uint8Array"))
+		return promise
+	}
+
+	// Obtain the underlying ArrayBuffer from the Uint8Array
+	ab, ok := intoObj.Get("buffer").Export().(goja.ArrayBuffer)
+	if !ok {
+		reject(newFsError(TypeError, "read() failed; reason: into argument must be a Uint8Array"))
+		return promise
+	}
+
+	// To avoid concurrency linked to modifying the runtime's `into` buffer from multiple
+	// goroutines we make sure to work on a separate copy, and will copy the bytes back
+	// into the runtime's `into` buffer once the promise is resolved.
+	intoBytes := ab.Bytes()
+	buffer := make([]byte, len(intoBytes))
+
+	// We register a callback to be executed by the VU's runtime.
+	// This ensures that the modification of the JS runtime's `into` buffer
+	// occurs on the main thread, during the promise's resolution.
+	callback := f.vu.RegisterCallback()
+	go func() {
+		n, readErr := f.file.Read(buffer)
+		callback(func() error {
+			_ = copy(intoBytes[0:n], buffer)
+
+			// Read was successful, resolve early with the number of
+			// bytes read.
+			if readErr == nil {
+				resolve(n)
+				return nil
+			}
+
+			var fsErr *fsError
+			isFSErr := errors.As(readErr, &fsErr)
+
+			if !isFSErr {
+				reject(readErr)
+				return nil
+			}
+
+			if fsErr.kind == EOFError && n == 0 {
+				resolve(goja.Null())
+			} else {
+				resolve(n)
+			}
+
+			return nil
+		})
+	}()
+
+	return promise
+}
+
+// Seek seeks to the given `offset` in the file, under the given `whence` mode.
+//
+// The returned promise resolves to the new `offset` (position) within the file, which
+// is expressed in bytes from the selected start, current, or end position depending
+// the provided `whence`.
+func (f *File) Seek(offset goja.Value, whence goja.Value) *goja.Promise {
+	promise, resolve, reject := f.vu.Runtime().NewPromise()
+
+	if common.IsNullish(offset) {
+		reject(newFsError(TypeError, "seek() failed; reason: the offset argument cannot be null or undefined"))
+		return promise
+	}
+
+	var intOffset int64
```

Comment:
Nit: the validation has a repetitive pattern that could be converted into a small abstraction, improving the code readability by reducing the method size

Like:

```golang

intOffset, err := extractInt(offset, rt)
if err != nil {
   reject(newFsError(TypeError, "seek() failed; reason: the offset argument " + err.Error()))
}

intWhence, err := extractInt(whence, rt)
if err != nil {
   reject(newFsError(TypeError, "seek() failed; reason: the whence argument " + err.Error()))
}

func extractInt(v goja.Value, rt *goja.Runtime) (int64, error) {
   if common.IsNullish(v) {
       return 0, errors.New("is null or undefined")
   }
   
   var intV int64
   if err := f.vu.Runtime().ExportTo(offset, &intV); err != nil {
      return 0, errors.New("cannot be interpreted as an integer")
   }

   return intV, nil
}
```

Reply:
👍🏻 

I've applied the pattern you suggested in https://github.com/grafana/k6/commit/ac8a5a7deb2f9e70b1fbc00a6dc3f679bdbd9494